libs/pika/executors/include/pika/executors/thread_pool_scheduler_queue_bypass.hpp

```diff
@@ -0,0 +1,275 @@
+        using thread_pool_scheduler::operator!=;
+
+        // Since this scheduler executes the task immediately on whichever thread is active
+        // we do not support "with_priority" property
+        // we do not support "with_hint" property
+
+        // support with_stacksize property
+        friend thread_pool_scheduler_queue_bypass tag_invoke(
+            pika::execution::experimental::with_stacksize_t,
+            thread_pool_scheduler_queue_bypass const& scheduler,
+            pika::execution::thread_stacksize stacksize)
+        {
+            auto sched_with_stacksize = scheduler;
+            sched_with_stacksize.stacksize_ = stacksize;
+            return sched_with_stacksize;
+        }
+
+        friend pika::execution::thread_stacksize tag_invoke(
+            pika::execution::experimental::get_stacksize_t,
+            thread_pool_scheduler_queue_bypass const& scheduler)
+        {
+            return scheduler.stacksize_;
+        }
+
+        // support with_annotation property
+        friend constexpr thread_pool_scheduler_queue_bypass tag_invoke(
+            pika::execution::experimental::with_annotation_t,
+            thread_pool_scheduler_queue_bypass const& scheduler, char const* annotation)
+        {
+            auto sched_with_annotation = scheduler;
+            sched_with_annotation.annotation_ = annotation;
+            return sched_with_annotation;
+        }
+
+        friend thread_pool_scheduler_queue_bypass tag_invoke(
+            pika::execution::experimental::with_annotation_t,
+            thread_pool_scheduler_queue_bypass const& scheduler, std::string annotation)
+        {
+            auto sched_with_annotation = scheduler;
+            sched_with_annotation.annotation_ =
+                pika::detail::store_function_annotation(PIKA_MOVE(annotation));
+            return sched_with_annotation;
+        }
+
+        // support get_annotation property
+        friend constexpr char const* tag_invoke(pika::execution::experimental::get_annotation_t,
+            thread_pool_scheduler_queue_bypass const& scheduler) noexcept
+        {
+            return scheduler.annotation_;
+        }
+
+        template <typename F>
+        void execute(F&& f, char const* fallback_annotation) const
+        {
+            using namespace pika::debug::detail;
+            using namespace pika::threads::detail;
+            using namespace pika::detail;
+
+            if (get_self_id() != invalid_thread_id)
+            {
+                throw std::runtime_error("Bypass scheduler - already on task thread");
+                return;
+            }
+
+            // which thread index are we in the current pool
+            const std::int16_t thread_num = get_local_thread_num_tss();
+            // create a description object with annotation
+            thread_description desc(
+                f, (fallback_annotation != nullptr) ? fallback_annotation : "Bypass");
```

Comment:
I think the fallback annotation should be non-null. Maybe replace with an assert and 
```suggestion
            thread_description desc(f, fallback_annotation);
```
?